### PR TITLE
Detect when type/field declared in an interface shadows a member from base interface.

### LINKED ISSUE #38711: Warning is not reported for a type shadowing a declaration from base interface.
```

        [Fact]
        public void NestedTypes_33()
        {
            var source1 =
@"
interface I1
{
    interface I2
    { }
}

interface I3 : I1
{
    interface I2
    {
    }
}

class C1
{
    public interface I4
    { }
}

class C3 : C1
{
    public interface I4
    {
    }
}
";

            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
                                                 parseOptions: TestOptions.Regular);

            compilation1.VerifyDiagnostics(
                // (23,22): warning CS0108: 'C3.I4' hides inherited member 'C1.I4'. Use the new keyword if hiding was intended.
                //     public interface I4
                Diagnostic(ErrorCode.WRN_NewRequired, "I4").WithArguments("C3.I4", "C1.I4").WithLocation(23, 22)
                );
        }
```

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #38711.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

File: src/Compilers/CSharp/Portable/Symbols/OverriddenOrHiddenMembersHelpers.cs [hunk 1]
```diff
@@ -518,84 +518,111 @@ private static OverriddenOrHiddenMembersResult MakeInterfaceOverriddenOrHiddenMe
...
-                    {
-                        currTypeHasSameKindNonMatch = true;
+                            break;
+                        case SymbolKind.NamedType:
+                            if (otherMember.GetMemberArity() == memberArity)
+                            {
+                                currTypeHasExactMatch = true;
+                                currTypeBestMatch = otherMember;
+                            }
+                            break;
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Symbols/OverriddenOrHiddenMembersHelpers.cs]: Nit: blank line below, but no blank line on the previous case.

**Pull request merged**